### PR TITLE
main group filling order

### PR DESCRIPTION
**Pull request by luvvlyjude**
resource pack option to fill the main group in different orders, random, and backwards

i wanted to add this because then if i do 3x3 and 9 max queued then it looks like how classic wall would look with randomly previews appearing.

### CODE REVIEW

**Review by KingContaria — CHANGES_REQUESTED** [review]

**Comment by sathya-pramodh** [comment]
Goat!!!!!

File: src/main/java/me/contaria/seedqueue/customization/Layout.java [hunk 1]
```diff
@@ -71,8 +73,8 @@ private static Layout fromJson(JsonObject jsonObject) throws JsonParseException
                 Group.fromJson(jsonObject.getAsJsonObject("main"), SeedQueue.config.rows, SeedQueue.config.columns),
                 jsonObject.has("locked") ? Group.fromJson(jsonObject.getAsJsonObject("locked")) : null,
                 jsonObject.has("preparing") ? Group.fromJson(jsonObject.getAsJsonArray("preparing")) : new Group[0],
-                jsonObject.has("replaceLockedInstances") && jsonObject.get("replaceLockedInstances").getAsBoolean()
-        );
+                jsonObject.has("replaceLockedInstances") && jsonObject.get("replaceLockedInstances").getAsBoolean(),
+                jsonObject.has("mainFillOrder") ? MainFillOrder.valueOf(jsonObject.get("mainFillOrder").getAsString().toUpperCase()) : MainFillOrder.FORWARD);
```
Comment (KingContaria) [inline src/main/java/me/contaria/seedqueue/customization/Layout.java:77]: `String#toUpperCase` is sensitive to localization, please use `toUpperCase(Locale.ROOT)` instead

File: src/main/java/me/contaria/seedqueue/gui/wall/SeedQueueWallScreen.java [hunk 2]
```diff
@@ -359,10 +360,19 @@ private void updateMainPreviews() {
         }
 
         this.preparingPreviews.sort(Comparator.comparing(SeedQueuePreview::isPreviewReady, Comparator.reverseOrder()));
-        for (int i = 0; i < this.mainPreviews.length && !this.preparingPreviews.isEmpty(); i++) {
-            if (SeedQueue.config.waitForPreviewSetup && !this.preparingPreviews.get(0).isPreviewReady()) {
+
+        ArrayList<Integer> previewsOrder = IntStream.range(0, this.mainPreviews.length).collect(ArrayList::new, List::add, List::addAll);
```
Comment (KingContaria) [inline src/main/java/me/contaria/seedqueue/gui/wall/SeedQueueWallScreen.java:364]: Please use fastutils `IntArrayList` here

**Review by KingContaria — CHANGES_REQUESTED** [review]

**Review by KingContaria — APPROVED** [review]
Lgtm, thank you! Will hold off on merging until i start work on the 1.4 update

**Comment by luvvlyjude** [comment]
yay yay yippee iyipeeeeee I did it thanks I appreciate i thanks 😊 ☺

**Pull request merged**